manager/integration/tests/test_zone.py

```diff
@@ -492,6 +494,95 @@ def test_replica_auto_balance_zone_best_effort(client, core_api, volume_name):
     assert z3_r_count == 2
 
 
+def test_replica_auto_balance_when_replica_on_unschedulable_node(client, core_api, volume_name, request):  # NOQA
+    """
+    Scenario: replica auto-balance when replica already running on
+              an unschedulable node.
+
+    Issue: https://github.com/longhorn/longhorn/issues/4502
+
+    Given set `replica-soft-anti-affinity` to `true`.
+    And set `replica-zone-soft-anti-affinity` to `true`.
```

Comment:
Why we need to make both `soft-anti-affinity` to `true`?

That's because replica auto-rebalancing will only take effect when node/zone or both soft-anti-affinity, because replicas could be possible to stay on the same node/zone, then they will be rebalanced based on the `replicaAutoBalance` setting.